rs/rosetta-api/icp_ledger/rosetta-integration-tests/tests/tests.rs

```diff
@@ -846,3 +857,86 @@ async fn test_query_block_range() {
 
     assert_eq!(response.blocks.len(), 10);
 }
+
+#[tokio::test]
+async fn test_block_transaction() {
+    let env = TestEnv::setup_or_panic(true).await;
+    assert!(env
+        .rosetta
+        .block_transaction(
+            BlockIdentifier {
+                index: 100,
+                hash: "INVALID_HASH".to_owned()
+            },
+            TransactionIdentifier {
+                hash: "INVALID_TX_HASH".to_owned()
+            }
+        )
+        .await
+        .unwrap_err()
+        .0
+        .message
+        .contains("Block not found"));
+
+    let minter = test_identity()
+        .sender()
+        .expect("test identity sender not found!");
+    let mut block_indices: Vec<Nat> = vec![];
+    for i in 0..4u64 {
+        let mint_arg = TransferArg {
+            from_subaccount: None,
+            to: Account::from(Principal::anonymous()),
+            fee: None,
+            created_at_time: None,
+            memo: None,
+            amount: Nat::from(i),
+        };
+        let arg = Encode!(&mint_arg).unwrap();
+        block_indices.push(
+            env.pocket_ic
+                .update_call(env.ledger_id, minter, "icrc1_transfer", arg)
+                .await
+                .expect("Unable to submit call")
+                .unwrap_as::<Result<BlockIndex, TransferError>>()
+                .expect("Error performing icrc1_transfer"),
+        );
```

Comment:
Does this create a Rosetta block? From my (limited) understanding from previous discussions, I would have thought that this creates four blocks (with different timestamps), each with just a single transaction?

Reply:
It does create RosettaBlocks, however it is possible that each RosettaBlock only stores one transaction. This depends on the execution ticker of pocket ic. I will change this to be deterministic.

Reply:
Thanks! Ah, this is where the `live` <-> `deterministic` dance is needed with pocket IC and Rosetta.